### PR TITLE
[TS migration] Migrate AnchorForAttachmentsOnly component

### PR DESCRIPTION
**Pull request by ruben-rebelo**
### Details
[TS migration] Migrate _**AnchorForAttachmentsOnly**_ component to TypeScript

### Fixed Issues
$ https://github.com/Expensify/App/issues/25121
PROPOSAL: N/A

### Tests
- [x] Verify that no errors appear in the JS console

**Test AnchorForAttachmentsOnly**:

1. Make sure attachments looks as expected on the chat
2. Upload attachments of different extensions
3. Attachments should work as before depending on the extension
4. Attachments are downloadable

### Offline tests
N/A

### QA Steps
- [x] Verify that no errors appear in the JS console

Same as in the **Tests** section.

### PR Author Checklist
- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/Expensify/App/assets/39693995/1b9da1f5-2083-4047-85e0-a58adb43d05e

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/Expensify/App/assets/39693995/1929f9e3-32d2-4d78-8361-120e174af734

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/Expensify/App/assets/39693995/0fd3c521-49d9-40e0-bbfa-40df0ab6b697

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/Expensify/App/assets/39693995/8de3a690-5296-45ee-ac0d-fc6d682b6169

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/Expensify/App/assets/39693995/0cc78584-c284-4436-a9da-c84329175c40

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/Expensify/App/assets/39693995/8276b40e-3bbe-4885-b9b5-3c0f4277e549

</details>

### CODE REVIEW

File: src/components/AnchorForAttachmentsOnly/AnchorForAttachmentsOnlyTypes.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import {StyleProp, ViewStyle} from 'react-native';
+
+type AnchorForAttachmentsOnlyProps = {
+    source: string;
```
Comment (VickyStash) [inline src/components/AnchorForAttachmentsOnly/AnchorForAttachmentsOnlyTypes.ts:4]: Please add comments, you can reuse the one from removed [anchorForAttachmentsOnlyPropTypes.js](https://github.com/Expensify/App/pull/33429/files#diff-c1f581eca6e59cae5ca37b65ab3cc176b90ee3adac31047df5b341679ac21dc3) file

File: src/components/AnchorForAttachmentsOnly/AnchorForAttachmentsOnlyTypes.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+    displayName: string;
+    style: StyleProp<ViewStyle>;
```
Comment (VickyStash) [inline src/components/AnchorForAttachmentsOnly/AnchorForAttachmentsOnlyTypes.ts:6]:
```suggestion
    style?: StyleProp<ViewStyle>;
```

File: src/components/AnchorForAttachmentsOnly/BaseAnchorForAttachmentsOnly.tsx [hunk 3]
```diff
@@ -10,37 +9,28 @@ import * as ReportUtils from '@libs/ReportUtils';
...
-    onPressOut: PropTypes.func,
-
-    /** If a file download is happening */
-    download: PropTypes.shape({
-        isDownloading: PropTypes.bool.isRequired,
-    }),
-
-    ...anchorForAttachmentsOnlyPropTypes,
+type BaseAnchorForAttachmentsOnlyPropsWithOnyx = {
+    download: OnyxEntry<OnyxDownload>;
```
Comment (VickyStash) [inline src/components/AnchorForAttachmentsOnly/BaseAnchorForAttachmentsOnly.tsx:16]:
```suggestion
    /** If a file download is happening */
    download: OnyxEntry<OnyxDownload>;
```

File: src/components/AnchorForAttachmentsOnly/BaseAnchorForAttachmentsOnly.tsx [hunk 4]
```diff
@@ -10,37 +9,28 @@ import * as ReportUtils from '@libs/ReportUtils';
...
-    download: {isDownloading: false},
-    ...anchorForAttachmentsOnlyDefaultProps,
-};
+type BaseAnchorForAttachmentsOnlyProps = {
+    /** Press in handler for the link */
+    onPressIn?: () => void;
+    /** Press out handler for the link */
+    onPressOut?: () => void;
+} & AnchorForAttachmentsOnlyProps &
+    BaseAnchorForAttachmentsOnlyPropsWithOnyx;
```
Comment (VickyStash) [inline src/components/AnchorForAttachmentsOnly/BaseAnchorForAttachmentsOnly.tsx:25]:
```suggestion
type BaseAnchorForAttachmentsOnlyProps = AnchorForAttachmentsOnlyProps &
BaseAnchorForAttachmentsOnlyPropsWithOnyx &  {
    /** Press in handler for the link */
    onPressIn?: () => void;
    
    /** Press out handler for the link */
    onPressOut?: () => void;
};
```

File: src/components/AnchorForAttachmentsOnly/BaseAnchorForAttachmentsOnly.tsx [hunk 5]
```diff
@@ -10,37 +9,28 @@ import * as ReportUtils from '@libs/ReportUtils';
...
 
-function BaseAnchorForAttachmentsOnly(props) {
+function BaseAnchorForAttachmentsOnly(props: BaseAnchorForAttachmentsOnlyProps) {
```
Comment (VickyStash) [inline src/components/AnchorForAttachmentsOnly/BaseAnchorForAttachmentsOnly.tsx:27]: Please destructure the props

**Comment by ruben-rebelo** [comment]
I have read the CLA Document and I hereby sign the CLA

File: src/components/AnchorForAttachmentsOnly/AnchorForAttachmentsOnlyTypes.ts
Comment (fabioh8010) [inline src/components/AnchorForAttachmentsOnly/AnchorForAttachmentsOnlyTypes.ts:1]: Let's rename the file to just `types.ts` to be aligned with the guidelines!

File: src/components/AnchorForAttachmentsOnly/AnchorForAttachmentsOnlyTypes.ts [hunk 6]
```diff
@@ -0,0 +1,12 @@
+import {StyleProp, ViewStyle} from 'react-native';
+
+type AnchorForAttachmentsOnlyProps = {
+    /** The URL of the attachment */
+    source: string;
+    /** Filename for attachments. */
+    displayName: string;
+    /** Any additional styles to apply */
+    style?: StyleProp<ViewStyle>;
+};
```
Comment (fabioh8010) [inline src/components/AnchorForAttachmentsOnly/AnchorForAttachmentsOnlyTypes.ts:10]:
```suggestion
type AnchorForAttachmentsOnlyProps = {
    /** The URL of the attachment */
    source?: string;

    /** Filename for attachments. */
    displayName?: string;

    /** Any additional styles to apply */
    style?: StyleProp<ViewStyle>;
};
```

According to the propTypes.

File: src/components/AnchorForAttachmentsOnly/BaseAnchorForAttachmentsOnly.tsx [hunk 7]
```diff
@@ -10,57 +9,51 @@ import * as ReportUtils from '@libs/ReportUtils';
...
-const propTypes = {
-    /** Press in handler for the link */
-    onPressIn: PropTypes.func,
-
-    /** Press out handler for the link */
-    onPressOut: PropTypes.func,
+import {Download as OnyxDownload} from '@src/types/onyx';
+import AnchorForAttachmentsOnlyProps from './AnchorForAttachmentsOnlyTypes';
 
+type BaseAnchorForAttachmentsOnlyPropsWithOnyx = {
```
Comment (fabioh8010) [inline src/components/AnchorForAttachmentsOnly/BaseAnchorForAttachmentsOnly.tsx:15]:
```suggestion
type BaseAnchorForAttachmentsOnlyOnyxProps = {
```

File: src/components/AnchorForAttachmentsOnly/BaseAnchorForAttachmentsOnly.tsx [hunk 8]
```diff
@@ -10,57 +9,51 @@ import * as ReportUtils from '@libs/ReportUtils';
...
+    BaseAnchorForAttachmentsOnlyPropsWithOnyx & {
+        /** Press in handler for the link */
+        onPressIn?: () => void;
+        /** Press out handler for the link */
+        onPressOut?: () => void;
+    };
 
-function BaseAnchorForAttachmentsOnly(props) {
-    const sourceURL = props.source;
+function BaseAnchorForAttachmentsOnly({style, source, displayName, download, onPressIn, onPressOut}: BaseAnchorForAttachmentsOnlyProps) {
```
Comment (fabioh8010) [inline src/components/AnchorForAttachmentsOnly/BaseAnchorForAttachmentsOnly.tsx:28]:
```suggestion
function BaseAnchorForAttachmentsOnly({style, source = '', displayName = '', download, onPressIn, onPressOut}: BaseAnchorForAttachmentsOnlyProps) {
```

According to the deleted `src/components/AnchorForAttachmentsOnly/anchorForAttachmentsOnlyPropTypes.js` file.

File: src/components/AnchorForAttachmentsOnly/index.native.tsx [hunk 9]
```diff
@@ -1,21 +1,19 @@
...
 
-function AnchorForAttachmentsOnly(props) {
+function AnchorForAttachmentsOnly({source, displayName}: AnchorForAttachmentsOnlyTypes) {
     const styles = useThemeStyles();
     return (
         <BaseAnchorForAttachmentsOnly
-            // eslint-disable-next-line react/jsx-props-no-spreading
-            {...props}
+            source={source}
+            displayName={displayName}
```
Comment (fabioh8010) [inline src/components/AnchorForAttachmentsOnly/index.native.tsx:11]: What is the reason of this change? Is possible to keep old logic (spreading props)?

File: src/components/AnchorForAttachmentsOnly/index.tsx [hunk 10]
```diff
@@ -1,22 +1,21 @@
...
 
-function AnchorForAttachmentsOnly(props) {
+function AnchorForAttachmentsOnly({source, displayName, style}: AnchorForAttachmentsOnlyProps) {
     return (
         <BaseAnchorForAttachmentsOnly
-            // eslint-disable-next-line react/jsx-props-no-spreading
-            {...props}
+            source={source}
+            displayName={displayName}
+            style={style}
```
Comment (fabioh8010) [inline src/components/AnchorForAttachmentsOnly/index.tsx:12]: Same, is possible to keep old logic (spreading props)?

File: src/components/AnchorForAttachmentsOnly/index.native.tsx (hunk 9 above)
Comment (ruben-rebelo) [reply src/components/AnchorForAttachmentsOnly/index.native.tsx:11]: I did this change because I saw the rule of no spreading so decided to follow it.
I'll revert this change 👍

**Review by fabioh8010 — APPROVED** [review]

File: src/components/AnchorForAttachmentsOnly/BaseAnchorForAttachmentsOnly.tsx [hunk 11]
```diff
@@ -10,57 +9,52 @@ import * as ReportUtils from '@libs/ReportUtils';
...
+        onPressIn?: () => void;
+
+        /** Press out handler for the link */
+        onPressOut?: () => void;
+    };
 
-function BaseAnchorForAttachmentsOnly(props) {
-    const sourceURL = props.source;
+function BaseAnchorForAttachmentsOnly({style, source = '', displayName = '', download, onPressIn, onPressOut}: BaseAnchorForAttachmentsOnlyProps) {
+    const sourceURL = source;
```
Comment (BartoszGrajdek) [inline src/components/AnchorForAttachmentsOnly/BaseAnchorForAttachmentsOnly.tsx:30]: NAB: Do we really need this? I think it was used to not repeat `props.source` over and over

File: src/components/AnchorForAttachmentsOnly/BaseAnchorForAttachmentsOnly.tsx [hunk 12]
```diff
@@ -10,57 +9,52 @@ import * as ReportUtils from '@libs/ReportUtils';
...
     const sourceURLWithAuth = addEncryptedAuthTokenToURL(sourceURL);
-    const sourceID = (sourceURL.match(CONST.REGEX.ATTACHMENT_ID) || [])[1];
-    const fileName = props.displayName;
+    const sourceID = (sourceURL.match(CONST.REGEX.ATTACHMENT_ID) ?? [])[1];
+    const fileName = displayName;
```
Comment (BartoszGrajdek) [inline src/components/AnchorForAttachmentsOnly/BaseAnchorForAttachmentsOnly.tsx:33]: Same here ☝🏻

**Review by BartoszGrajdek — APPROVED** [review]
Small comments, but LGTM! 😄

File: src/components/AnchorForAttachmentsOnly/BaseAnchorForAttachmentsOnly.tsx (hunk 11 above)
Comment (ruben-rebelo) [reply src/components/AnchorForAttachmentsOnly/BaseAnchorForAttachmentsOnly.tsx:30]: It is done like that for a better default initialisation of displayName and source props.
I find it easier to identify the default value by destructuring the props.

File: src/components/AnchorForAttachmentsOnly/BaseAnchorForAttachmentsOnly.tsx (hunk 12 above)
Comment (ruben-rebelo) [reply src/components/AnchorForAttachmentsOnly/BaseAnchorForAttachmentsOnly.tsx:33]: You are correct, I'll push the changes

**Review by sobitneupane — APPROVED** [review]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>



https://github.com/Expensify/App/assets/25876548/063ed1e1-e921-470b-a586-3cd7f003c184


</details>

<details>
<summary>Android: mWeb Chrome</summary>



</details>

<details>
<summary>iOS: Native</summary>



https://github.com/Expensify/App/assets/25876548/9dfbe9f4-5879-402a-a73c-751dde4fd1af


</details>

<details>
<summary>iOS: mWeb Safari</summary>



https://github.com/Expensify/App/assets/25876548/ec7ad383-457f-4cab-ba5f-12fa8d035079


</details>

<details>
<summary>MacOS: Chrome / Safari</summary>



https://github.com/Expensify/App/assets/25876548/10ce1af0-ef9a-4454-909e-79863e15da0b


https://github.com/Expensify/App/assets/25876548/06dc8cd4-14cb-44ae-9dab-d724d5aa7d16


</details>

<details>
<summary>MacOS: Desktop</summary>



https://github.com/Expensify/App/assets/25876548/98795e8f-a2e9-4776-a24e-0548fd945054


</details>

**Review by sobitneupane — APPROVED** [review]

**Review by Gonals — DISMISSED** [review]

**Comment by Gonals** [comment]
Seems like we have a few issues, including unsigned commits and some linter problems

**Comment by VickyStash** [comment]
@Gonals oh, you are right! The author of the PR @ruben-rebelo is OOO till Monday, so I think it makes sense to get back to it after that, thank you!

**Comment by Gonals** [comment]
A few quick lint issues left!

**Comment by ruben-rebelo** [comment]
It should be good now @Gonals, Thank you!

**Comment by ruben-rebelo** [comment]
The branch wasn't synced with main and there were lint rules that have changes.
Sorry about it.
Now it's ready @Gonals.
Thank you!

**Comment by Gonals** [comment]
Run prettier and you are done! 😁

**Comment by ruben-rebelo** [comment]
@Gonals done

**Review by Gonals — APPROVED** [review]
There we go!
